lib/matplotlib/axes/_axes.py

```diff
@@ -4764,6 +4764,14 @@ def fill_between(self, x, y1, y2=0, where=None, interpolate=False,
                 for filling between two sets of x-values
 
         """
+        if not rcParams['_internal.classic_mode']:
+            color_aliases = mcoll._color_aliases
+            kwargs = cbook.normalize_kwargs(kwargs, color_aliases)
+
+            if not any(c in kwargs for c in ('color',) + tuple(color_aliases)):
```

Comment:
This is where edgecolor is treated equally with facecolor--it seems like a code shortcut that will yield a surprising result.